desktop_version/src/Entity.cpp

```diff
@@ -4389,8 +4391,8 @@ void entityclass::fixfriction( int t, float xfix, float xrate, float yrate )
     if (entities[t].vx > 6) entities[t].vx = 6.0f;
     if (entities[t].vx < -6) entities[t].vx = -6.0f;
 
-    if (std::abs(entities[t].vx-xfix) <= xrate) entities[t].vx = xfix;
-    if (std::abs(entities[t].vy) < yrate) entities[t].vy = 0;
+    if (std::fabs(entities[t].vx-xfix) <= xrate) entities[t].vx = xfix;
+    if (std::fabs(entities[t].vy) < yrate) entities[t].vy = 0;
```

Comment:
Why change this? Are you sure that we won't be taking out any rounding errors that resulted from the previous implicit conversion to int? (This also applies to the other places where you switched to using `fabs()`.)

Reply:
I assumed that any previous behavior was a bug.

Reply:
So let's think each case through...

- `entityclass::fixfriction()` is unused, so we can ignore that. That function should probably be removed.
- `entityclass::applyfriction()` is used for entities that have a true `gravity` attribute, which just means humanoids (player and crewmates), so it's most easily tested by checking player movement.

  The code here checks if the absolute value of the player's x-velocity, after rounding towards 0, is less than 1.1 (`xrate` will always be 1.1 here), and if so, it will set their x-velocity to 0. It also checks if the absolute value of the player's y-velocity, after rounding towards 0, is less than 0.25 (`yrate` will always be 0.25), and if so, it will set their y-velocity to 0 as well. What changing this to `fabs` or `fabsf` does is take out the rounding towards 0.

  The player's x-velocity depends upon if they're holding left or right. If they're holding left or right for at least 5 frames, they will undergo deceleration when they release the directional key; else they will simply stop in place and have their x-velocity set to 0.

  So on the first frame of holding left or right, the initial value of the player's x-velocity is 3 - then `applyfriction()` will subtract 1.1 from that, leaving their x-velocity at 1.9. 1.9 is not less than 1.1, but 1.9 rounded to 1 *is* less than 1.1, so this is a bit worrying. However, their x-velocity would have been set to 0 anyways because they held the directional key for less than 5 frames, so this doesn't matter.

  So what if the player has been holding their directional key for 5 frames, and then lets go, starting to decelerate? Well, they are guaranteed to have an x-velocity of 6 at this point (because their acceleration x-velocities per frame after holding a directional key would have been 1.9 -> 3.8 -> 5.7 -> 6, so they would be guaranteed to have full speed at this point), and here is where `applyfriction()` comes in to decelerate them, subtracting 1.1 from their x-velocity every frame. This means their speed on each frame will be 6 -> 4.9 -> 3.8 -> 2.7 -> 1.6 -> 0.5 -> 0.

  But wait, this is where the rounding comes in. The function performs the check in question *after* subtracting from their x-velocity, so when the function subtracts from their x-velocity to make it 1.6, it will check: is 1.6, after rounding towards 0, less than 1.1? And 1.6 rounded towards 0 is 1, which *is* less than 1.1.

  So that means their velocity each frame on the current build is 6 -> 4.9 -> 3.8 -> 2.7 -> 0.

  But if you change it to `fabs`/`fabsf` and take out the rounding, then the function will no longer zero out their velocity when it gets to 1.6. This PR makes it so that each frame is now 6 -> 4.9 -> 3.8 -> 2.7 -> 1.6 -> 0.

  This essentially results in having an extra frame of deceleration when letting go of a directional key after holding it for 5 frames. This means you have added one extra frame of moving 2 pixels to the left, if decelerating to the left, and added one extra frame of moving 1 pixel to the right, if decelerating to the right. (The discrepancy between 2 and 1 is because of a further round towards zero when computing the `newxp` and `newyp` of the player.) Which means...

  _This is a physics change!_ **In an unrelated PR!** _**Why did you do this, Leo?!**_
- The activity zone and teleporter prompt checks both check if the player x-velocity is less than or equal to 1, after rounding it towards 0.

  As covered above, on the first frame of the player's acceleration, their x-velocity is 1.9. 1.9, after rounding this towards 0, is equal to 1. But if you take out the rounding, then 1.9 is *not* less than or equal to 1. So taking out the rounding means that you'd no longer be able to activate activity zones or teleporter prompts on the first frame of acceleration, which is yet another physics change, in a PR that's primarily about something else.

In conclusion, swapping these functions to `fabs`/`fabsf` is most likely going to break some of my TASes that synced fine on 2.2. And also, don't put unrelated changes in your PR.

Reply:
Oh, huh... we should probably document that it's abs for a reason (an unintended one, admittedly). In any case, it'd be fine to just submit the compile fix separately, then this PR can pull from master and I'll squash it all down when it's ready.

Reply:
Is this even a compile fix? There's tons of implicit conversions like this between float and int all around the codebase, and the most they do is emit warnings on MSVC (which should be disabled, I can probably grab a snippet from LodePNG that does that). That's why I'm so baffled Leo included this.

Reply:
It might be a CFLAG - it builds on flibitBuild but I have a habit of instinctively adding -std flags, and Cent GCC default's to a very old, very strict standard.

Reply:
I see. All the more reason to switch to using `SDL_abs()` instead, then. But I agree with submitting the switch to `SDL_abs()` separately.

Reply:
#575

Reply:
I initially resolved this, but it'd make more sense to wait for #575 to be merged (at which point I can do a rebase).

Reply:
Merged that, thanks!